### PR TITLE
Feat/add catalog i18n

### PR DESCRIPTION
**Pull request by mario-mui**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mario-mui** [comment]
@benjdlambert please review this PR 😁

File: plugins/catalog/src/components/EntityOrphanWarning/EntityOrphanWarning.tsx [hunk 1]
```diff
@@ -53,8 +56,7 @@ export function EntityOrphanWarning() {
   return (
     <>
       <Alert severity="warning" onClick={() => setConfirmationDialogOpen(true)}>
-        This entity is not referenced by any location and is therefore not
-        receiving updates. Click here to delete.
+        {t('deleteEntity.description')}
```
[the remark this answers is not in the document]
Comment (mario-mui) [reply plugins/catalog/src/components/EntityOrphanWarning/EntityOrphanWarning.tsx:59]: sorry, I'm not get. you mean a key like `t('description')` ?

**Comment by mario-mui** [comment]
ping

**Comment by freben** [comment]
Hey, sorry but a rebase is needed

**Comment by mario-mui** [comment]
> Hey, sorry but a rebase is needed

done

**Comment by benjdlambert** [comment]
@mario-mui the build is red at the minute, looks like somethings need fixing up :pray:

**Comment by mario-mui** [comment]
@benjdlambert I had fixed all actions expect `Accessibility`, it seems not my fault

**Comment by mario-mui** [comment]
ping @benjdlambert

**Comment by mario-mui** [comment]
@benjdlambert Shall we start review this PR, 😄

**Comment by mario-mui** [comment]
> @mario-mui sorry - been a busy few weeks so not had chance to get round to this yet!

sure, can someone else review this pr ？

**Comment by mattemoore** [comment]
Any way a non-contributor can help on this PR?  Really need this to satisfy official languages support for adoption in Canadian government.

**Comment by mario-mui** [comment]
> I see there are conflicts now.
> 
> We'll get to it as soon as possible.

ok，today I will fix conflicts

File: plugins/catalog-react/src/components/FavoriteEntity/FavoriteEntity.tsx [hunk 2]
```diff
@@ -51,7 +54,11 @@ export const FavoriteEntity = (props: FavoriteEntityProps) => {
       onClick={() => toggleStarredEntity()}
     >
       <Tooltip
-        title={isStarredEntity ? 'Remove from favorites' : 'Add to favorites'}
+        title={
+          isStarredEntity
+            ? t('favoriteEntity.RemoveFromFavorites')
```
Comment (freben) [inline plugins/catalog-react/src/components/FavoriteEntity/FavoriteEntity.tsx:59]:
```suggestion
            ? t('favoriteEntity.removeFromFavorites')
```

File: plugins/catalog-react/src/translation.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    entitySearchBarPlaceholder: 'Search',
+    entityTagPickerTitle: 'Tags',
+    entityTypePicker: {
+      title: 'Type',
+      errorMessage: 'Failed to load entity types',
+      optionAllTitle: 'all',
+    },
+    favoriteEntity: {
+      addToFavorites: 'Add to favorites',
+      RemoveFromFavorites: 'Remove from favorites',
```
Comment (freben) [inline plugins/catalog-react/src/translation.ts:52]:
```suggestion
      removeFromFavorites: 'Remove from favorites',
```

File: plugins/catalog-react/src/translation.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    entityPeekAheadPopover: {
+      title: 'Drill into the entity to see all of the tags.',
+      emailCardAction: {
+        title: 'Email {{email}}',
+        subTitle: 'mailto {{email}}',
+      },
+      entityCardActionsTitle: 'Show details',
+    },
+    entityProcessingStatusPickerTitle: 'Processing Status',
+    entitySearchBarPlaceholder: 'Search',
```
Comment (freben) [inline plugins/catalog-react/src/translation.ts:43]: Is this a pattern we've adopted, that if there's just one string at this point in time we skip the dot? I thought maybe these would have been like `entitySearchBar: { placeholder:` to facilitate adding more fields in the future.

If this is how we've done it before, and if it's standard i18n structure, then i'm fine either way.

**Review by freben — COMMENTED** [review]
Awesome! Just small questions

File: plugins/catalog-react/src/translation.ts (hunk 4 above)
Comment (mario-mui) [reply plugins/catalog-react/src/translation.ts:43]: make sense. we should use `object structure`, so  i18n keys are more extendable

**Review by freben — APPROVED** [review]
Excellent!

**Pull request merged**